demos/python_demos/multi_camera_multi_person_tracking/multi_camera_multi_person_tracking.py

```diff
@@ -81,74 +118,120 @@ def run(params, capture, detector, reid):
         if frames is None:
             continue
 
-        all_detections = detector.get_detections(frames)
+        frame_number += 1
+        all_detections = detector.wait_and_grab()
+        detector.run_async(frames, frame_number)
+
         all_masks = [[] for _ in range(len(all_detections))]
         for i, detections in enumerate(all_detections):
             all_detections[i] = [det[0] for det in detections]
             all_masks[i] = [det[2] for det in detections if len(det) == 3]
 
-        tracker.process(frames, all_detections, all_masks)
+        tracker.process(prev_frames, all_detections, all_masks)
         tracked_objects = tracker.get_tracked_objects()
 
-        fps = round(1 / (time.time() - start), 1)
-        vis = visualize_multicam_detections(frames, tracked_objects, fps)
+        latency = time.time() - start
+        avg_latency.update(latency)
+        fps = round(1. / latency, 1)
+
+        vis = visualize_multicam_detections(prev_frames, tracked_objects, fps, **config['visualization_config'])
         if not params.no_show:
             cv.imshow(win_name, vis)
-            if cv.waitKey(1) == 27:
-                break
+
         if output_video:
             output_video.write(cv.resize(vis, video_output_size))
 
+        print('\rProcessing frame: {}, fps = {} (avg_fps = {:.3})'.format(
+                            frame_number, fps, 1. / avg_latency.get()), end="")
+        prev_frames, frames = frames, prev_frames
+    print('')
+
     thread_body.process = False
     frames_thread.join()
 
     if len(params.history_file):
         history = tracker.get_all_tracks_history()
+        save_dir = os.path.dirname(params.history_file)
+        if not os.path.exists(save_dir):
+            os.makedirs(save_dir)
```

Comment:
This fails for me if I run the demo with `--history_file a.txt`

Reply:
Fixed.